pkg/apis/pipeline/v1/pipeline_validation.go

```diff
@@ -269,6 +271,22 @@ func (pt *PipelineTask) validateWorkspaces(workspaceNames sets.String) (errs *ap
 	return errs
 }
 
+// validateEnabledInlineSpec validates that pipelineSpec or taskSpec is allowed by checking
+// disable-inline-spec field
+func (pt PipelineTask) validateEnabledInlineSpec(ctx context.Context) (errs *apis.FieldError) {
+	if pt.TaskSpec != nil {
+		if config.FromContextOrDefaults(ctx).FeatureFlags.DisableInlineSpec {
+			errs = errs.Also(apis.ErrDisallowedFields("taskSpec"))
+		}
+	}
+	if pt.PipelineSpec != nil {
```

Comment:
All embedded Spec can be disabled now including the one for Pipeline in Pipeline.